src/components/main.js

```diff
@@ -0,0 +1,16 @@
+import React from "react";
+
+export const Main = props => {
+    let welcomeMsg;
+
+    try {
+        const {email} = props.history.location.state;
+        welcomeMsg = `Hello ${email}`
+    } catch (e) {
+        if (e instanceof TypeError) {
```

Comment:
why are you specifically catching `TypeError` ? 

Maybe what you want is to check first if `props.history` and `props.history.location` exist, if so, then you can extract email from them.

the try, catch here is not needed

Reply:
Oh I see , At first I thought it'd be better this way to make it readable. It's now reworked to have those nested ifs .

Reply:
You have a good point about the overkilled if's. Actually some tools like Typescript or coffescript provide you sugar syntax for nesting null checks (example in Typescript, `props.history?.location?.state || 'default message' `)

Unfortunately we haven't added Typescript yet (it could be good to have though 🤔), even so it's better to add the check null for two reasons:

1. `try-catch` is commonly used for unexpected and out of our control problems, like Error from some dependency (external library), Network connection/communication, I/O files operation. First you should try to get your flow cases under controls like check nulls for variables.
2. You said `make it readable`. Uhmmm that could be false, since is not common to catch those cases but instead validating them through if-else sentences